### PR TITLE
[Merged by Bors] - feat(smartmodule): added `jolt` smartmodule

### LINKED ISSUE #376: Jolt SmartModule
Implement Jolt SmartModule publishable to Hub

### PR DESCRIPTION
**Pull request by galibey**
Fixes #376

### CODE REVIEW

File: smartmodules/jolt/hub/package-meta.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+package_format_version: "0.1"
+name: jolt
+version: 0.1.0
```
Comment (sehz) [inline smartmodules/jolt/hub/package-meta.yaml:4]: I think this file is not needed. It should be automatically generated by hub publish. 

Lets have separate issue to have release CI job for publishing

File: smartmodules/jolt/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+edition = "2021"
+license = "Apache-2.0"
+publish = false
+
+[lib]
+crate-type = ['cdylib', 'rlib']
+
+[dependencies]
+fluvio-jolt = { version = "0.1"}
+fluvio-smartmodule = { version = "0.3" }
```
Comment (sehz) [inline smartmodules/jolt/Cargo.toml:15]: Should be using latest fluvio smartmodule on master until we release it

File: smartmodules/jolt/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+fn init(params: SmartModuleExtraParams) -> Result<()> {
+    if let Some(raw_spec) = params.get("spec") {
+        match serde_json::from_str(raw_spec) {
+            Ok(spec) => {
+                SPEC.set(spec).expect("spec is already initialized");
+                Ok(())
+            }
+            Err(err) => {
+                eprintln!("unable to parse spec from params: {:?}", err);
+                Err(eyre::Report::msg(
```
Comment (sehz) [inline smartmodules/jolt/src/lib.rs:21]: This should return missing param error.  See regex example

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  Couple of nits

File: smartmodules/jolt/src/lib.rs (hunk 3 above)
Comment (galibey) [reply smartmodules/jolt/src/lib.rs:24]: Done

File: smartmodules/jolt/Cargo.toml (hunk 2 above)
Comment (galibey) [reply smartmodules/jolt/Cargo.toml:15]: It is using the latest from master. It's patched on workspace-level Cargo.toml

File: smartmodules/jolt/hub/package-meta.yaml (hunk 1 above)
Comment (galibey) [reply smartmodules/jolt/hub/package-meta.yaml:4]: It is generated with paths to the crate's `target` folder, but in our case, we need the workspace's `target` folder. So we need to manually modify this file.

Comment (sehz) [reply smartmodules/jolt/hub/package-meta.yaml:4]: Got it. @superstator, need to fix this https://github.com/infinyon/fluvio/issues/2652

File: smartmodules/jolt/Cargo.toml (hunk 2 above)
Comment (sehz) [reply smartmodules/jolt/Cargo.toml:15]: I was wondering about that.  Great.

**Comment by galibey** [comment]
bors+

**Pull request closed without merging**